cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -120,11 +131,29 @@ public boolean hasNamespace(Id.Namespace namespaceId) {
    * @param metadata the {@link NamespaceMeta} for the new namespace to be created
    * @throws AlreadyExistsException if the specified namespace already exists
    */
-  public void createNamespace(NamespaceMeta metadata) throws AlreadyExistsException {
+  public void createNamespace(NamespaceMeta metadata) throws AlreadyExistsException, IOException {
+    // TODO: CDAP-1427 - This should be transactional, but we don't support transactions on files yet
     NamespaceMeta existing = store.createNamespace(metadata);
     if (existing != null) {
       throw new AlreadyExistsException(NAMESPACE_ELEMENT_TYPE, metadata.getId());
     }
+    String namespaceId = metadata.getId();
+    Location namespaceLocation = locationFactory.create(metadata.getId());
+    String namespacePath = namespaceLocation.toURI().getPath();
+    if (namespaceLocation.exists()) {
+      // need this hack because default namespace cannot be deleted, so the directory is always found, once created
```

Comment:
I believe that this problem will still persist even when default_namespace creation is moved to a more deterministic time.